### PR TITLE
Style fixes

### PR DESCRIPTION
**Pull request by dimagalat**
pylab causes issues with reproducibility and is best avoided. Info on side effects: https://matthiasbussonnier.com/posts/10-No-PyLab-Thanks.html
a few other small issues like mixed quote characters are fixed

### CODE REVIEW

**Comment by dimagalat** [comment]
else was replaced with elif to avoid a potential bug

**Comment by dimagalat** [comment]
added tree visualisation to improve clarity

**Comment by rogelj** [comment]
Thanks @dimagalat - I will take a look at the revisions and merge if no conflict. @jeff-boykin let me know if you want to take a look too.

File: curriculum/02-materials/code/DataScience101_Part2_DecisionTree.ipynb [hunk 1]
```diff
@@ -300,6 +315,23 @@
    "source": [
     "get_code(IrisTree, iris_data.columns)"
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can also graphically represent the resulting tree:"
```
Comment (rogelj) [inline curriculum/02-materials/code/DataScience101_Part2_DecisionTree.ipynb]: Hey @dimagalat - can you put a note here to clarify that Graphviz needs to be installed for the tree to be rendered?

**Comment by dimagalat** [comment]
Fair point. Note added

**Comment by dimagalat** [comment]
@rogelj please review when you get a chance

**Comment by dimagalat** [comment]
I forgot to mention the fact that I've updated notebooks to python 3, since many core libraries are stopping support for python 2 this year, with the rest to follow next year.

**Comment by rogelj** [comment]
Thanks @dimagalat. The changes have been approved.

**Pull request merged**